sdk/azcore/response.go

```diff
@@ -84,12 +91,19 @@ func (r *Response) UnmarshalAsByteArray(v **[]byte, format Base64Encoding) error
 // UnmarshalAsJSON calls json.Unmarshal() to unmarshal the received payload into the value pointed to by v.
 // If no payload was received a RequestError is returned.  If json.Unmarshal fails a UnmarshalError is returned.
 func (r *Response) UnmarshalAsJSON(v interface{}) error {
```

Comment:
Can you make sure we have a test that unmarshalls JSON (or XML, doesn't really matter) when the body download policy is disabled?  I suspect we don't test this today.

Reply:
Done